src/core/Form/FilterInput/FilterInput.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { Component, ChangeEvent } from 'react';
+import { default as styled } from 'styled-components';
+import { withSuomifiDefaultProps } from '../../theme/utils';
+import {
+  HtmlInput,
+  HtmlInputProps,
+  HtmlDiv,
+  HtmlDivProps,
+} from '../../../reset';
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { baseStyles } from './FilterInput.baseStyles';
+import classnames from 'classnames';
+import { Omit } from '../../../utils/typescript';
+import { LabelText, LabelMode } from '../LabelText/LabelText';
+import { AutoId } from '../../../utils/AutoId';
+import { InputStatus } from '../types';
+import { StatusText } from '../StatusText/StatusText';
+import { getConditionalAriaProp } from '../../../utils/aria';
+
+const baseClassName = 'fi-filter-input';
+const filterInputClassNames = {
+  error: `${baseClassName}--error`,
+  disabled: `${baseClassName}--disabled`,
+  labelAlignLeft: `${baseClassName}--label-align-left`,
+  wrapper: `${baseClassName}_wrapper`,
+  functionalityContainer: `${baseClassName}_functionalityContainer`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  inputElement: `${baseClassName}_input`,
+};
+
+type FilterInputStatus = Exclude<InputStatus, 'success'>;
+
+export interface FilterInputProps<T>
+  extends Omit<HtmlInputProps, 'type'>,
+    TokensProp {
+  /** FilterInput container div class name for custom styling. */
+  className?: string;
+  /** FilterInput container div props */
+  inputContainerProps?: Omit<HtmlDivProps, 'className'>;
+  /** Disable input usage */
+  disabled?: boolean;
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: LabelMode;
+  /** Text to mark a field optional. Will be wrapped in parentheses and shown after labelText. */
+  optionalText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: FilterInputStatus;
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** FilterInput name */
+  name?: string;
+  /** Align label on top or on the left side of the input field
+   * @default 'top'
+   */
+  labelAlign?: 'top' | 'left';
+  /** Items to be filtered */
+  items: Array<T>;
+  /** Returns the filtered items */
+  onFiltering: (filteredItems: Array<T>) => void;
+  /** Filtering rule to be used */
+  filterRule: (item: T, query: string) => boolean;
```

Comment:
And also for this, I'd prefer something that tells the user this is a function / callback rather than e.g. a reg.exp. rule. What about filterFunction or filterFunc? The functionality of these callbacks needs to be communicated through naming in my opinion.

Reply:
maybe call these onFilter and filterFunc, that would be clear for me at least. And it's different from onChange which could also refer to changes in the search term / input value.

Reply:
My thoughts as well. Is Func a commonly used abbreviation when it comes to these?  I'd maybe just use 'filterFunction' to be explicit.

Reply:
Changed `filterRule` to `filterFunc` as discussed.